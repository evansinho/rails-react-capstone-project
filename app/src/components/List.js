import React, { useEffect, useState, useRef } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as houseActions from '../actions/houseActions';
import Nav from './Nav';
import Carousel from './Carousel';
import list from '../styles/List.module.css';
import Loader from './Loader';

const List = ({ actions, state }) => {
  const parent = useRef(null);

  useEffect(() => {
    const token = localStorage.getItem('token');
    actions.fetchSubscription('https://aqueous-wildwood-18424.herokuapp.com/meets', {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    });
  }, []);

  const [navOpen, setNavOpen] = useState(false);

  const { subscription } = state;

  const clickHandler = () => {
    if (navOpen) {
      setNavOpen(false);
    }
  };

  const KeyDownHandler = () => {
    clickHandler();
  };

  return (
    <div
      onClick={clickHandler}
      onKeyDown={KeyDownHandler}
      role="button"
      ref={parent}
      className={list.wrapper}
      aria-label="Close side navigation menu"
      tabIndex={-1}
    >
      <Nav open={navOpen} setNavOpen={setNavOpen} parent={parent} searchOpen={false} local="Meets" />
      <main className={list.main}>
        {subscription
          ? <Carousel count={subscription.length} list={subscription} />
          : <Loader />}
      </main>
    </div>
  );
};

function mapStateToProps({ state }) {
  return { state: { ...state } };
}

function mapActionsToProps(dispatch) {
  return {
    actions: bindActionCreators({ ...houseActions }, dispatch),
  };
}

List.propTypes = {
  actions: PropTypes.objectOf(PropTypes.any).isRequired,
  state: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default connect(mapStateToProps, mapActionsToProps)(withRouter(List));
