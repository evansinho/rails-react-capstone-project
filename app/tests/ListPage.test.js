import React from 'react';
import renderer from 'react-test-renderer';
import List from '../src/containers/List';
import Wrapper from './helpers/Wrapper';

test('List is rendered', () => {
  const component = renderer.create(
    <Wrapper>
      <List />
    </Wrapper>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});