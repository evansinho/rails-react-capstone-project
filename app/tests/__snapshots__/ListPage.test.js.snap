// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List is rendered 1`] = `
<div
  aria-label="Close side navigation menu"
  className="wrapper"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={-1}
>
  <nav
    className="main"
  >
    <ul>
      <li>
        <i
          aria-label="Open navigation menu"
          className="fas fa-bars"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={-1}
        />
      </li>
      <li>
        <h1
          className="title"
        >
          Meets
        </h1>
      </li>
      <li>
        <input
          className="search-nav hidden"
          onKeyDown={[Function]}
          placeholder="Search..."
          type="text"
        />
        <i
          aria-label="Search"
          className="fas fa-search search-ico"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={-1}
        />
      </li>
    </ul>
  </nav>
  <main
    className="main"
  >
    <div
      className="main"
    >
      <a
        className="main"
        href="/"
      >
        <svg
          className="circle"
          height="50"
          viewBox="0 0 100 100"
          width="50"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <linearGradient
              id="gradient"
              x1="0%"
              x2="0%"
              y1="0%"
              y2="100%"
            >
              <stop
                offset="0%"
                stopColor="#ec5200"
              />
              <stop
                offset="100%"
                stopColor="transparent"
              />
            </linearGradient>
          </defs>
          <circle
            cx="50"
            cy="50"
            fill="none"
            r="40"
            stroke="url(#gradient)"
            strokeWidth="13"
            transform="rotate(90 50 50)"
          />
        </svg>
        <div
          className="text-container"
        >
          <span
            className="text--light"
          >
            Circle
          </span>
        </div>
      </a>
    </div>
  </main>
</div>
`;
